src/tests/test_process.py

```diff
@@ -32,10 +54,27 @@ class TestProcessProto(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         """Setup environment and start pybitmessage"""
-        cls.home = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        cls.home = set_appdata_folder(tempfile.gettempdir())
+        cls._cleanup_files()
         put_signal_file(cls.home, 'unittest.lock')
+        starttime = time.time()
         subprocess.call(cls._process_cmd)  # nosec
-        time.sleep(5)
+        timeout = starttime + 30.0
+        while time.time() <= timeout:
+            if os.path.exists(os.path.join(cls.home,
+                                           'singleton.lock')):
+                try:
+                    pstat = os.stat(os.path.join(cls.home, 'singleton.lock'))
+                    if starttime <= pstat.st_mtime and pstat.st_size > 0:
+                        break
+                except OSError:
+                    break
+            time.sleep(1)
+        if time.time() >= timeout:
+            raise psutil.TimeoutExpired("Timeout starting {}".format(cls._process_cmd))
+        # wait a bit for the program to fully start
+        # 10 sec should be enough
+        time.sleep(10)
```

Comment:
Why sleep 10 seconds if singleton.lock is already checked? What may happen?

Reply:
OK, I see. It is to start all threads and not fail test_threads and test_shutdown.

Reply:
But I hope that 5 sec will be enough for any environment (on travis it works without delay). This delay is used before starting tests in any test case subclassed from `TestProcessProto` on each pass.